### ISSUE TITLE
CakePHP 3.0 - Simple belongsToMany....

### ISSUE THREAD

**Issue by Laoei** [cakephp/cakephp#6976]
Hi everyone, i'm using Cakephp 3.0 since few weeks and i've got a problem with the relation belongsToMany... :>

## Model : TournamentsTable.php
    class TournamentsTable extends Table {
	public function initialize(array $config) {		
		$this->belongsToMany('Teams');
	    }
    }
## Model : TeamsTable.php
    class TeamsTable extends Table {
	public function initialize(array $config) {
		$this->belongsToMany('Tournaments');
	    }
    }
## View : add.tcp
    echo $this->Form->create($tournament);
    echo $this->Form->input('name');
    echo $this->Form->input('teams._ids', ['options' => $teams, 'multiple' => 'checkbox']);
    echo $this->Form->button(__('Add'));
    echo $this->Form->end();
## Controller : TournamentsController.php
    public function add() {
        if ($this->request->is('post')) {
            $tournament = $this->Tournaments->newEntity($this->request->data);
            if($this->Tournaments->save($tournament)) {
                return $this->redirect(['controller' => 'anyController', 'action' => 'index']);
            } else {
			...
            }
        }
    }

## Errors

Warning (2): array_combine(): Both parameters should have an equal number of elements [CORE\src\ORM\Association\BelongsToMany.php, line 734]

Warning (4096): Argument 1 passed to Cake\Database\Expression\QueryExpression::_addConditions() must be of the type array, boolean given, called in ........\cakephp\cakephp\src\Database\Expression\QueryExpression.php on line 124 and defined [CORE\src\Database\Expression\QueryExpression.php, line 503]

Warning (2): Invalid argument supplied for foreach() [CORE\src\Database\Expression\QueryExpression.php, line 509]

....

Cannot set an empty property

...

Please try correcting the issue for the following table aliases:

TeamsTournaments

## debug($this->request->data())
    [
        'name' => 'Tournament01',
        'teams' => [
            '_ids' => [
                (int) 0 => '7',
                (int) 1 => '8'
            ]
        ]
    ]

Someone can help me ? plz :dancers: :8ball:

**Comment by Laoei** [comment]
Database deleted.
New Database.
=> Add is woriking for Tournaments but not the linking table (TeamsTournaments : team_tournaments in mysql)

**Comment by markstory** [comment]
Does your entity have `teams` in the accessible fields list?

**Comment by Laoei** [comment]
I added this :

    protected $_accessible = [
            'teams' => true
    ];

Now, i have an error =P

"Cannot insert row, some of the primary key values are missing. Got (, ), expecting (id, games_id)"

## View : add.tcp
    $this->Form->input('games_id', ['value' => '2'])

**Comment by Laoei** [comment]
debug($tournament->id); // null
    debug($tournament->games_id); // null

**Comment by lorenzo** [comment]
@Laoei It seems like you have a table with `$this->primaryKey(['id', 'game_id'])` Are you sure that is what you need?

**Comment by Laoei** [comment]
id : Tournament.id
    game_id : Tournament.game_id (foreign key)

## debug($this->request->data());
    [
        'name' => 'test',
        'games_id' => '2',
        'Teams' => [
            '_ids' => [
                (int) 0 => '1;1',
                (int) 1 => '2;1'
            ]
        ]
    ]

**Comment by lorenzo** [comment]
I don't understand your answer @Laoei 

Do you have a table with that composite key?

**Comment by Laoei** [comment]
I've got this (in mysql) :

    tournaments
        id
        name
        picture
        game_id

**Comment by lorenzo** [comment]
please dump your database schema so we can look at at you have. Also showing the `initialize()` function of all your tables would help

**Comment by Laoei** [comment]
I made my database with "mysql workbench".
It's mysql workbench the problem... I d'ont master the tool lol

It's working since this tool, thx for helping

Closed issue.

**Comment by thomaslener** [comment]
"Cannot insert row, some of the primary key values are missing. Got (, ), expecting (id, games_id)"

Just solved this error by editing the ExampleTable.php:

Caused the error:
$this->primaryKey(['id', 'something_id']);

changed to this and it works:
$this->primaryKey('id');

I created the database using mysql workbench, and then did cake bake. So if you did like i did, the workaround shown above might work for you too.
